tests/lib/external/snapd-testing-tools/tools/tests.systemd

```diff
@@ -59,23 +59,51 @@ stop_and_remove_unit() {
 }
 
 wait_for_service() {
-    local service_name="$1"
-    local state="${2:-active}"
+     if [ $# -eq 0 ]; then
+        show_help
+        exit 0
+    fi
+
+    local attempts=300
+    local wait=1
+    local state="active"
+    local service_name
+
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            --wait)
+                wait="$2"
+                shift 2
+                ;;
+            --n)
+                attempts="$2"
+                shift 2
+                ;;
+            --state)
+                state="$2"
+                shift 2
+                ;;
+            *)
+                service_name=$1
```

Comment:
Not sure if shellcheck will complain about this but
```suggestion
                service_name="$1"
```

Reply:
it shouldn't be needed because it gets the first parameter.